### ISSUE TITLE
mat-contrast sass mixin provides failing accessibility colors

### ISSUE THREAD

**Issue by wratho** [angular/components#19123]
#### Bug, feature request, or proposal:
When using mat-contrast in sass from angular/material/theming the color provided doesn't pass accessibility standards for WCAG AA for normal text. Also doesn't follow the color guidelines from the material color tool

#### What is the expected behavior?
Provide color that meets accessibility standards

#### What is the current behavior?
doesn't

#### What are the steps to reproduce?
https://angular-ujpp3f.stackblitz.io
you can see in this example of utilizing mat-teal 600 and the given mat-contrast you can key in the values here: https://monsido.com/tools/contrast-checker and see that it fails AA standard for normal text.

#### What is the use-case or motivation for changing an existing behavior?
It feels misleading to have this functionality and for it to not pass web standards.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
angular/material 7-9

#### Is there anything else we should know?
This is not limited to all colors, but each palette seems to have at least one that is wrong.

**Comment by jelbourn** [comment]
(revisiting older issues as part of a triaging fix-it)

All of the palette colors and their contrast colors are pulled directly from the 2014 - 2016 era Material Design spec ([found here](https://material.io/design/color/the-color-system.html#tools-for-picking-colors)). We will likely update these at some point, but for the time being we don't have a great replacement.

**Comment by amysorto** [comment]
As of v17.2 we have support for M3 for our components. Closing this since this issue is outdated now.
